app/controllers/users_controller.rb

```diff
@@ -0,0 +1,46 @@
+class UsersController < ApplicationController
+  before_action :set_user, only: [:show, :edit, :update, :destroy]
+  before_action :ensure_superuser
+
+  # GET /users
+  def index
+    @users = User.all
+  end
+
+  # GET /users/1
+  def show
+  end
+
+  # GET /users/1/edit
+  def edit
+  end
+
+  # PATCH/PUT /users/1
+  def update
+    if params[:toggle]
+      if @user.has_role? :admin
+        @user.remove_role :admin
+      else
+        @user.add_role :admin
+      end
+    end
+    redirect_to users_path
+  end
+
+  # DELETE /users/1
+  def destroy
+    @user.destroy
+    redirect_to users_url, notice: 'User was successfully destroyed.'
+  end
+
+  private
+    # Use callbacks to share common setup or constraints between actions.
+    def set_user
+      @user = User.find(params[:id])
```

Comment:
I think we need to use a try catch here, because if someone tries to query user with `id` not in the database, it will throw an error and render error page. So, it would be better if we prevent to show that page by rendering 404 not found.

Reply:
Done